core/src/test/java/io/questdb/cairo/BitmapIndexTest.java

```diff
@@ -660,6 +660,82 @@ public void testCppLatestByIndexReaderIndexedWithTruncate() throws Exception {
         });
     }
 
+    @Test
+    public void testCppLatestByIndexReaderIgnoreSparseKeysUpdates() {
+        final int valueBlockCapacity = 32;
+        final int keyCount = 42;
+
+        create(configuration, path.trimTo(plen), "x", valueBlockCapacity);
+
+        try (BitmapIndexWriter writer = new BitmapIndexWriter(configuration, path, "x")) {
+            for (int i = 0; i < keyCount; i++) {
+                // add even keys
+                if(i % 2 == 0) {
+                    writer.add(i, i);
+                }
+            }
+        }
+
+        DirectLongList rows = new DirectLongList(keyCount, MemoryTag.NATIVE_LONG_LIST);
```

Comment:
this and the reader should be closed in try() ... finally block to avoid leakage on test failure

Reply:
fixed